src/main/java/hudson/remoting/PingThread.java

```diff
@@ -61,14 +61,24 @@
      */
     private final long interval;
 
-    public PingThread(Channel channel, long timeout, long interval) {
+    /**
+     * Specifis if troubleshooting information should be catched in case of pingFailure
+     */
+    private final boolean pingFailureAnalyzer;
+
+    public PingThread(Channel channel, long timeout, long interval, boolean pingFailureAnalyzer) {
         super("Ping thread for channel "+channel);
         this.channel = channel;
         this.timeout = timeout;
         this.interval = interval;
+        this.pingFailureAnalyzer = pingFailureAnalyzer;
         setDaemon(true);
     }
 
+    public PingThread(Channel channel, long timeout, long interval) {
+        this(channel, 4*60*1000/*4 mins*/, interval, false);
```

Comment:
:bug: this should pass the timeout supplied and not replace it with something else...